src/main/java/put/io/black/java/ScenarioGUI.java

```diff
@@ -68,17 +127,24 @@ public void mouseClicked(MouseEvent e) {
                 } catch (NumberFormatException err) {
                     JOptionPane.showMessageDialog(null, "Liczba poziomów nie jest liczbą!");
                 }
-
             }
         });
     }
 
-
+    /**
+     * Main class to run GUI.
+     * @param args for now args are for nothing.
+     */
     public static void main(String args[]) {
         JFrame frame = new JFrame("ScenarioGUI");
         frame.setContentPane(new ScenarioGUI().panelMain);
         frame.setDefaultCloseOperation(JFrame.EXIT_ON_CLOSE);
-        frame.pack();
         frame.setVisible(true);
     }
+
+    private String stringToUrl(String input) throws MalformedURLException, URISyntaxException {
+        URL url= new URL(input);
+        URI uri = new URI(url.getProtocol(), url.getUserInfo(), url.getHost(), url.getPort(), url.getPath(), url.getQuery(), url.getRef());
```

Comment:
Can you check UriBuilder or OkHttp and tell what do you think about using it in the project?